### PR TITLE
@ashfurrow => Add screen properties

### PR DESCRIPTION
**Pull request by alloy**
I _think_ I’ve done all the groundwork needed to implement screen views with properties. Once https://github.com/artsy/eigen/pull/454 is done, then this should be good to be merged.

### CODE REVIEW

File: ARAnalyticalProvider.m [hunk 1]
```diff
@@ -62,7 +62,19 @@ - (void)logTimingEvent:(NSString *)event withInterval:(NSNumber *)interval prope
...
+- (void)didShowNewPageView:(NSString *)pageTitle withProperties:(NSDictionary *)properties {
+    NSDictionary *props;
+    if (properties.count > 0) {
+        NSMutableDictionary *merge = [properties mutableCopy];
+        merge[@"screen"] = pageTitle;
+        props = [merge copy];
+    } else {
+        props = @{ @"screen":pageTitle };
+    }
+    [self event:@"Screen view" withProperties:props];
```
Comment (orta) [inline ARAnalyticalProvider.m]: can we get super properties merged in here too

Comment (orta) [inline ARAnalyticalProvider.m]: https://github.com/orta/ARAnalytics/blob/master/ARAnalytics.m#L500-L501

File: ARAnalytics.m [hunk 2]
```diff
@@ -537,6 +537,18 @@ + (void)pageView:(NSString *)pageTitle
     }];
 }
 
++ (void)pageView:(NSString *)pageTitle withProperties:(NSDictionary *)properties
+{
+    if (!pageTitle) return;
```
Comment (ashfurrow) [inline ARAnalytics.m]: @orta do we want to do this? Maybe an assert instead?

**Comment by ashfurrow** [comment]
Looks good to me. @orta you have a note about super properties. How about I merge and then submit a new PR adding those? I'd PR into Alloy's branch, but :baby:

**Comment by orta** [comment]
Sure, I agree.

**Comment by ashfurrow** [comment]
Cooool.

File: ARAnalytics.m (hunk 2 above)
Comment (orta) [inline ARAnalytics.m]: Happy with an assert ( double check if thats a pattern we use in ARAnalytics please)

Comment (ashfurrow) [inline ARAnalytics.m]: Yeah good idea.

Comment (ashfurrow) [inline ARAnalytics.m]: Nah, no asserts anywhere else.

**Pull request merged**